shopfloor_base/actions/base_action.py

```diff
@@ -31,6 +31,8 @@ def get_actions_for(
         and hasattr(component_instance.work, attr_name)
     }
     kwargs.update(kw)
+    if component_instance.collection._name == "shopfloor.app":
+        kwargs["app"] = component_instance.collection
```

Comment:
I would add an explicit arg  `app=None` and then
```suggestion
    if not app and component_instance.collection._name == "shopfloor.app":
        app = component_instance.collection
     kwargs["app"] = app
```

Then you can pass the app also from the actions class.
And we should probably add it to `_propagate_kwargs`.

You could also add `work_context.app` in the `__init___` of the base service class.

Reply:
I changed it.